app/main.py

```diff
@@ -1,2 +1,59 @@
+from typing import Any, Hashable
+
+
 class Dictionary:
-    pass
+    class Node:
+        def __init__(self, key: Hashable, value: Any) -> None:
+            self.key = key
+            self.value = value
+            self.hash = hash(key)
+
+    def __init__(self, initial_capacity: int = 8) -> None:
+        self.capacity = initial_capacity
+        self.size = 0
+        self.load_factor = 2 / 3
+        self.hash_table = [None] * self.capacity
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        if self.size / self.capacity > self.load_factor:
```

Comment:
The condition for resizing the hash table should be 'if self.size >= self.capacity * self.load_factor:' to ensure the table is resized when the load factor is exceeded.